arch/x86/boot/compressed/sl_stub.S

```diff
@@ -168,9 +168,25 @@ SYM_FUNC_START(sl_stub)
 	jmp	.Lcpu_setup_done
 
 .Ldo_unknown_cpu:
-	/* AMD is not yet supported */
+	/* Neither Intel nor AMD */
 	ud2
 
+.Ldo_amd:
+	cmpl	$(AMD_CPUID_MFGID_EBX), %ebx
+	jnz	.Ldo_unknown_cpu
+	cmpl	$(AMD_CPUID_MFGID_EDX), %edx
+	jnz	.Ldo_unknown_cpu
+	cmpl	$(AMD_CPUID_MFGID_ECX), %ecx
+	jnz	.Ldo_unknown_cpu
+
+	/* Know it is AMD */
+	movl	$(SL_CPU_AMD), sl_cpu_type(%ebp)
+
+	/*
+	 * Enable global interrupts including SMI and NMI (GIF).
+	 */
+	stgi
```

Comment:
One final thought. Did you want to move this until after the measurements in sl_main()? I am planning to move the instruction to enable SMI until the end of that too. @dpsmith  was saying we should keep SMI disabled until after the measurements are done.

Reply:
I am not sure about that. On the one hand it would prevent SMM code from messing up communication with TPM, on the other hand, we shouldn't trust such SMM if it does.

IMHO it would be better to delay this until interrupt handlers are installed, as it enables not only SMI.

Reply:
I agree with you first point on SMM and your second point I think is a larger discussion. We can just leave it there for now if you want to get this in...

Reply:
I'm not devoted to keeping it here, we can move it. Lets keep it as similar as possible for Intel and AMD.

Reply:
For now what you have is inline with what TXT is doing. So let's just keep it here for now.